examples/layeredeffects/ID1.fst

```diff
@@ -226,7 +203,6 @@ let tot_i #a (f : unit -> Tot a) : I a =
 let i_tot #a (f : unit -> I a) : Tot a =
   reify (f ()) (fun _ -> True) ()
 
-[@@expect_failure [19]]
```

Comment:
This used to fail earlier due to unifier generating smt guards when applying indexed effects combinators. E.g. `f #t e =?= f #(x:t{phi}) e`, which resulted in a `forall (x:t). phi` guard that was unprovable. With substitutive indexed effects, all of this doesn't come up.

There are many such instances where I could remove the `expected_failure` attribute.